structs.go

```diff
@@ -743,6 +743,129 @@ type GuildPreview struct {
 	Description string `json:"description"`
 }
 
+// GuildScheduledEvent is a representation of a scheduled event in a guild. Only for retrieval of the data.
+// https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event
+type GuildScheduledEvent struct {
+	// The ID of the scheduled event
+	ID string `json:"id"`
+	// The guild id which the scheduled event belongs to
+	GuildID string `json:"guild_id"`
+	// The channel id in which the scheduled event will be hosted, or null if scheduled entity type is EXTERNAL
+	ChannelID string `json:"channel_id"`
+	// The id of the user that created the scheduled event
+	CreatorID string `json:"creator_id"`
+	// The name of the scheduled event (1-100 characters)
+	Name string `json:"name"`
+	// The description of the scheduled event (1-1000 characters)
+	Description string `json:"description"`
+	// The time the scheduled event will start
+	ScheduledStartTime time.Time `json:"scheduled_start_time"`
+	// The time the scheduled event will end, required only when entity_type is EXTERNAL
+	ScheduledEndTime *time.Time `json:"scheduled_end_time"`
+	// The privacy level of the scheduled event
+	PrivacyLevel GuildScheduledEventPrivacyLevel `json:"privacy_level"`
+	// The status of the scheduled event
+	Status GuildScheduledEventStatus `json:"status"`
+	// See the requirement fields
+	// https://discord.com/developers/docs/resources/guild-scheduled-event#guild-scheduled-event-object-field-requirements-by-entity-type
+	EntityType GuildScheduledEventEntityType `json:"entity_type"`
+	// The id of an entity associated with a guild scheduled event
+	EntityID string `json:"entity_id"`
+	// Additional metadata for the guild scheduled event
+	EntityMetadata GuildScheduledEventEntityMetadata `json:"entity_metadata"`
+	// The user that created the scheduled event
+	Creator *User `json:"creator"`
+	// The number of users subscribed to the scheduled event
+	UserCount int `json:"user_count"`
+	// The cover image hash of the scheduled event
+	// see https://discord.com/developers/docs/reference#image-formatting for more
+	// information about image formatting
+	Image string `json:"image"`
+}
+
+// GuildScheduledEventParams are the parameters allowed for creating a new scheduled event.
+// or updating an existing scheduled event.
+// https://discord.com/developers/docs/resources/guild-scheduled-event#create-guild-scheduled-event
+type GuildScheduledEventParams struct {
+	// The channel id in which the scheduled event will be hosted, or null if scheduled entity type is EXTERNAL
+	ChannelID string `json:"channel_id,omitempty"`
```

Comment:
Okay, now this is going to be tricky. I think this struct also might require custom `MarshalJSON`. Let me explain.
When we're upgrading to external type, we need to explicitly set it to `null` (i.e. have it as `*string` without `omitempty`).
But, when we just want to update voice/stage event data, without specifying the channel, it's going to be a problem, because it will give us a error.

So I propose that you make a custom marshal function, which will marshal this field normally, with omitempty when `entity_type != "external"`, and otherwise (you can also add `&& params.ChannelID != ""` so it returns a error when the field is not empty and `entity_type = "external"`) will overwrite it with a nil pointer one, like that:
```go
json.Marshal(struct {
    GuildScheduledEventParams
    ChannelID *string `json:"channel_id"` // null
}{ GuildScheduledEventParams: params })

Reply:
I will follow the custom Marshall way, but like another feature of discord want to `null` statement, I have add a `NullableType` implementation of String and let possibility to add other Type later.

Let me know if you have feedback :)